pallets/pools/src/lib.rs

```diff
@@ -160,6 +169,17 @@ type LookUpSource<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::So
 // Type that indicates a point in time
 type Moment = u64;
 
+// Types to ease function signatures
+type PoolDetailsOf<T> = PoolDetails<
```

Comment:
Yay, I was going to make a pass at doing this in one of my PRs soon, updating all the generics everywhere was getting to be such a pain